api.py

```diff
@@ -870,6 +870,35 @@ def create_api_key(**kwargs):
     return json.loads(response.content)['apikey']
 
 
+def delete_api_key(key, **kwargs):
+    """
+    Delete an API key from this user's account.
+
+    Send an HTTP POST request to delete the given API key.  The key in question may not be used for authentication of the request.
+
+    Parameters
+    ----------
+    key
+        The string representation of the API key to delete
+    **kwargs
+        See :py:func:`make_post_request`
+
+    Returns
+    -------
+    Nothing
+
+    Raises
+    ------
+    requests.HTTPError
+        Requests that result in an error raise an exception with information about the failure. See :py:func:`requests.Response.raise_for_status` for more information.
+    """
+    if key == kwargs.get('api_key'):
```

Comment:
I suggest we remove this logic and allow users to use an API key to delete/remove the same API key.

Reply:
I put in this restriction (it doesn't exist in the REST API) to avoid accidents.  I'm happy to take it out if you're comfortable with the possibility of users accidentally deleting the wrong key.  Would it be acceptable to remove this check from this function but keep it in the command-line version?